packages/grid/x-grid/src/tests/filtering.XGrid.test.tsx

```diff
@@ -198,6 +198,39 @@ describe('<XGrid /> - Filter', () => {
     expect(getColumnValues()).to.deep.equal(['Adidas', 'Puma']);
   });
 
+  it('should trigger onFilterModelChange when the link operator changes', () => {
+    const onFilterModelChange = spy();
+    const newModel: GridFilterModel = {
+      items: [
+        {
+          columnField: 'brand',
+          value: 'a',
+          operatorValue: 'startsWith',
+        },
+        {
+          columnField: 'brand',
+          value: 'a',
+          operatorValue: 'endsWith',
+        },
+      ],
+    };
+    render(
+      <TestCase
+        filterModel={newModel}
+        onFilterModelChange={onFilterModelChange}
+        state={{
+          preferencePanel: { openedPanelValue: GridPreferencePanelsValue.filters, open: true },
+        }}
+      />,
+    );
+    expect(onFilterModelChange.callCount).to.equal(4);
+
+    const select = document.getElementById('columns-filter-operator-select');
```

Comment:
Side note:  we should be using `screen.getByRole('combobox', {name: /Operators/i})`. But, we can't, because the `htmlFor` prop is not set in the label, like recommended: https://next.material-ui.com/components/selects/#native-select.

<img width="500" src="https://user-images.githubusercontent.com/42154031/119024439-716bbe80-b979-11eb-8a6d-f8f692dd6eeb.png" alt="image" style="max-width:100%;">

Reply:
Pull the changes from master and you can apply the following suggestion: https://github.com/mui-org/material-ui-x/pull/1727

```suggestion
    const select = screen.queryAllByRole('combobox', { name: /Operators/i })[1];
```